src/main/resources/public/js/components/autosuggests/geoarea-autosuggest.js

```diff
@@ -58,20 +58,29 @@ class GeoAreaAutosuggest extends Component {
     };
 
     onSuggestionSelected = (event, {suggestion}) => {
-        this.props.onGeoAreaSelected(suggestion);
+        this.props.onGeoAreaSelected(event, suggestion);
     };
 
-    getSuggestionValue = suggestion => suggestion.name || GeoAreaAutosuggest.defaultProps.value;
+    getSuggestionValue = suggestion => {
+        return suggestion.name || GeoAreaAutosuggest.defaultProps.value;
+    };
 
     shouldRenderSuggestions = (input) => {
         return input && (input.trim().length >= sizeQueryBeforeFetch);
     };
 
+    handleOnChange = (event, {newValue}) => {
+        GeoAreaAutosuggest.defaultProps.value = newValue;
```

Comment:
A default prop should not change, it is only meant to hold a default value when none has been specified in props.value